packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -138,3 +151,77 @@ class ProcessLister {
     throw StateError('Failed to find $powershell or $pwsh on PATH');
   }
 }
+
+class WindowsVersionExtractor {
+  WindowsVersionExtractor(this.processManager);
+
+  final ProcessManager processManager;
+
+  Future<WindowsVersionExtractionResult> getDetails() async {
+    String? caption, releaseId, displayVersion;
+    try {
+      final ProcessResult captionResult = await processManager.run(
+        <String>['wmic', 'os', 'get', 'Caption,OSArchitecture'],
+      );
+
+      if (captionResult.exitCode == 0) {
+        final String? output = captionResult.stdout as String?;
+        if (output != null) {
+          final List<String> parts = output.split('\n');
+          if (parts.length >= 2) {
+            caption = parts[1].replaceAll('Microsoft Windows', '').replaceAll('  ', ' ').trim();
+          }
+        }
+      }
+    } on ProcessException {
+      // Ignored, use default null value.
+    }
+
+    try {
+      final ProcessResult osDetails = await processManager.run(
+        <String>['reg', 'query', r'HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion', '/t', 'REG_SZ'],
+      );
+
+      if (osDetails.exitCode == 0) {
+        final String? output = osDetails.stdout as String?;
+        if (output != null) {
+          final Map<String, String> data = Map<String, String>.fromEntries(
+              output.split('\n').where((String line) => line.contains('REG_SZ')).map((String line) {
+                final List<String> parts = line.split('REG_SZ');
+                return MapEntry<String, String>(parts.first.trim(), parts.last.trim());
+              }));
+          releaseId = data['ReleaseId'];
+          displayVersion = data['DisplayVersion'];
+        }
+      }
+    } on ProcessException {
+      // Ignored, use default null values.
```

Comment:
Perhaps we should log the error here. (See earlier comment https://github.com/flutter/flutter/pull/154797/files#r1805859210 for more)